### PR TITLE
feat: added the --firewall-driver option

### PR DESCRIPTION
**Pull request by unknowndevQwQ**
Change the firewall driver with the command option

Signed-off-by: unknowndevQwQ <unknowndevQwQ@pm.me>

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -12,6 +12,9 @@ struct Opts {
     /// Instead of reading from STDIN, read the configuration to be applied from the given file.
     #[clap(short, long)]
     file: Option<String>,
+    /// Select netavark's firewall driver
+    #[clap(short='i', long, env="NETAVARK_FW")]
```
Comment (Luap99) [inline src/main.rs:16]: just remove the short option, we do not use the cli directly so it serves little to no purpose anyway.

**Review by Luap99 — COMMENTED** [review]
Thanks for the contribution. 
We require a sign off as described here: https://github.com/containers/podman/blob/main/CONTRIBUTING.md#sign-your-prs
Also make sure to fix the validate issue from the CI logs, https://cirrus-ci.com/task/5515233758806016

File: src/main.rs (hunk 1 above)
Comment (unknowndevQwQ) [reply src/main.rs]: > just remove the short option, we do not use the cli directly so it serves little to no purpose anyway.

Currently netavark has no way of automatically completing parameters, short parameters may be useful in the future.

**Comment by unknowndevQwQ** [comment]
@Luap99 Is it okay now?

**Comment by Luap99** [comment]
Please squash your commits into one

File: src/main.rs (hunk 1 above)
Comment (Luap99) [reply src/main.rs]: Netavark is not intended to be ever executed manually on a CLI, I rather just have the long form. Using `-i` just fells random, who is going to remember what it means?

Comment (unknowndevQwQ) [reply src/main.rs]: > Netavark is not intended to be ever executed manually on a CLI, I rather just have the long form. Using `-i` just fells random, who is going to remember what it means?

I decided to give up until I found the right short parameters.

**Comment by unknowndevQwQ** [comment]
> Please squash your commits into one

I'm ready.

**Comment by unknowndevQwQ** [comment]
@Luap99 RUOK?

**Comment by maxxberg** [comment]
Any update?

**Comment by mheon** [comment]
We're finally getting around to getting the other firewall drivers plumbed in (expect the nftables driver to appear in a PR sometime next week, and firewalld will be finalized sometime after that), so if this is rebased I would be OK with merging

**Comment by unknowndevQwQ** [comment]
> We're finally getting around to getting the other firewall drivers plumbed in (expect the nftables driver to appear in a PR sometime next week, and firewalld will be finalized sometime after that), so if this is rebased I would be OK with merging

I need to spend some time generating an aesthetically pleasing gpg key, or rebase it later in github.dev

File: src/firewall/mod.rs [hunk 2]
```diff
@@ -45,13 +44,9 @@ enum FirewallImpl {
...
-
-    // If not requested lookup in NETAVARK_FW env var as well.
-    let driver = driver_name.or_else(|| env::var("NETAVARK_FW").ok());
-
-    if let Some(var) = driver {
-        debug!("Forcibly using firewall driver {}", var);
-        match var.to_lowercase().as_str() {
+    if let Some(driver) = driver_name {
+        debug!("Forcibly using firewall driver {driver}");
+        match driver.to_lowercase().as_str() {
```
Comment (mheon) [inline src/firewall/mod.rs:49]: Can we retain the environment variable here if driver_name is is none?

**Comment by mheon** [comment]
One change requested otherwise LGTM

File: src/firewall/mod.rs (hunk 2 above)
Comment (unknowndevQwQ) [reply src/firewall/mod.rs:49]: > Can we retain the environment variable here if driver_name is is none?

I'm already getting the value from the environment variable via clap's env attribute, so it should be consistent with the original behavior when no arguments are specified.

Comment (mheon) [reply src/firewall/mod.rs:49]: Clearly I need more coffee

**Comment by mheon** [comment]
@Luap99 PTAL, this should be ready

File: src/firewall/mod.rs (hunk 2 above)
Comment (unknowndevQwQ) [reply src/firewall/mod.rs:49]: > Clearly I need more coffee

But you don't look like you've been up all night or anything.

**Review by Luap99 — APPROVED** [review]
/lgtm

**Pull request merged**